atsign-foundation/at_server
# Fix for at_client_sdk-issue#384

**- What I did**
- Prevented the cached keys getting updated with empty value when refresh job runs.

**- How I did it**
- Add IF condition to check if value is empty. If empty, do not update the cached key value.

**- How to verify it**
Scenario - 1
1. Notify a key along with value with TTR set to -1
2. Reduce the corn job timer to 1 minute for Refresh Job.
3. When the cron job runs, the cached key created in step-1 should not be picked up.

Scenario - 2
1. Notify a key along with value with TTR set to 10 seconds
2. Reduce the corn job timer to 1 minute for Refresh Job.
3. When the cron job runs, the cached key value should not be updated to empty value.

**Additional context**
Fix for https://github.com/atsign-foundation/at_client_sdk/issues/384
